app/common/renderer/components/Session/AttachToSession.jsx

```diff
@@ -1,45 +1,43 @@
 import {ReloadOutlined} from '@ant-design/icons';
-import {Button, Card, Col, Form, Row, Select} from 'antd';
+import {Button, Card, Col, Form, Row, Select, Tooltip} from 'antd';
+import _ from 'lodash';
 import React from 'react';
 
 import {ServerTypes} from '../../actions/Session';
 import SessionStyles from './Session.module.css';
 
-const formatCaps = (caps) => {
-  let importantCaps = [caps.app, caps.platformName, caps.deviceName];
-  if (caps.automationName) {
-    importantCaps.push(caps.automationName);
+const formatCaps = (initialCaps, serverType) => {
+  let returnedCaps = [];
+  const caps =
+    serverType === ServerTypes.lambdatest && 'capabilities' in initialCaps
+      ? initialCaps.capabilities
+      : initialCaps;
+  // add sessionName (BrowserStack and other cloud providers only)
+  returnedCaps.push(caps.sessionName);
+  // add deviceName OR avd OR udid
+  const deviceName =
+    serverType === ServerTypes.lambdatest && 'desired' in caps
+      ? caps.desired.deviceName
+      : caps.deviceName;
+  returnedCaps.push(deviceName || caps.avd || caps.udid);
+  // add platformName and platformVersion
+  if (caps.platformName) {
+    const platformInfo = caps.platformVersion
+      ? `${caps.platformName} ${caps.platformVersion}`
+      : caps.platformName;
+    returnedCaps.push(platformInfo);
   }
-  return importantCaps.join(', ').trim();
+  // add automationName
+  returnedCaps.push(caps.automationName);
+  // add app OR bundleId OR appPackage
+  returnedCaps.push(caps.app || caps.bundleId || caps.appPackage);
+  // omit all values that were not found in caps
+  const nonEmptyCaps = _.reject(returnedCaps, _.isNil);
+  return nonEmptyCaps.join(' / ').trim();
 };
 
-const formatCapsBrowserstack = (caps) => {
-  let importantCaps = formatCaps(caps).split(', ');
-  if (caps.sessionName) {
-    importantCaps.push(caps.sessionName);
-  }
-  return importantCaps.join(', ').trim();
-};
-
-const formatCapsLambdaTest = (caps) => {
-  if (caps.hasOwnProperty.call(caps, 'capabilities')) {
-    caps = caps.capabilities;
-  }
-  const deviceName = caps.desired ? caps.desired.deviceName : caps.deviceName;
-  const importantCaps = [deviceName, caps.platformName, caps.platformVersion];
-  return importantCaps.join(', ').trim();
-};
-
-const getSessionInfo = (session, serverType) => {
```

Comment:
I would still prefer this format as it is more flexible and better readable in comparison to having to having all logic mixed into a single function. Vendor-specific stuff must be somehow separated

Reply:
Created separate methods for standard Appium server and cloud provider sessions. Although it's still a bit messy because some caps need specific handling (e.g. `deviceName`), so it ends up being less complicated to just duplicate most of the code.